addons/block_code/ui/main_panel.gd

```diff
@@ -139,17 +139,14 @@ func save_script():
 		_current_block_code_node.block_script = block_script
 		undo_redo.add_do_property(_current_block_code_node, "block_script", _current_block_code_node.block_script)
 
-	undo_redo.add_undo_property(_current_block_code_node.block_script, "block_trees", _current_block_code_node.block_script.block_trees)
-	undo_redo.add_undo_property(_current_block_code_node.block_script, "generated_script", _current_block_code_node.block_script.generated_script)
-
-	var block_trees := _block_canvas.get_canvas_block_trees()
+	_block_canvas.rebuild_block_trees(undo_redo)
 	var generated_script = _block_canvas.generate_script_from_current_window(block_script)
-	block_script.block_trees = block_trees
-	block_script.generated_script = generated_script
+	if generated_script != block_script.generated_script:
+		undo_redo.add_undo_property(_current_block_code_node.block_script, "generated_script", _current_block_code_node.block_script.generated_script)
+		block_script.generated_script = generated_script
+		undo_redo.add_do_property(_current_block_code_node.block_script, "generated_script", _current_block_code_node.block_script.generated_script)
```

Comment:
We shouldn't be changing the generated_script property on our own here, and we should consistently use the `var block_script: BlockScriptData` we defined earlier:

```suggestion
	if generated_script != block_script.generated_script:
		undo_redo.add_undo_property(block_script, "generated_script", block_script.generated_script)
		undo_redo.add_do_property(block_script, "generated_script", generated_script)
```

Reply:
Turns out we're doing this in a lot of places. I included this and a bunch of similar fixes in #140.